### ISSUE TITLE
Issue with Popup Components Rendering Behind Map Before Triggered by Marker Click

### ISSUE THREAD

**Issue by lukaslovre** [GrayFrost/sveaflet#16]
Hi,

I'm encountering an issue with the `Popup` component in my Svelte app. I'm not sure if this is due to an error on my part or if it's a bug.

#### Problem:
When the page initially loads, all components placed inside the `Popup` slots are rendered "behind" the map. When I click on a marker to open its corresponding popup, it removes the component from behind the map (and the DOM).

####  Explanation:
Here is my map code:

```svelte
{#each $properties as property (property.id)}
  <Marker
    latlng={new LatLng(property.lat, property.lng)}
    bind:instance={markerInstances[property.id]}
  >
    <Popup>
      <PropertyCard {property} />
    </Popup>
  </Marker>
{/each}
```

When the page loads, the HTML structure appears like this:

```html
<div class="leaflet-container">
  <div class="leaflet-pane leaflet-map-pane">...</div>
  <div class="leaflet-control-container">...</div>
  <!-- Multiple divs with PropertyCard components for each Popup defined -->
  <div>...</div>
</div>
```

#### Observations:
- Upon initial load, there are `div` elements corresponding to each `PropertyCard` component within the popup.
- When I click on a marker and its popup opens, the corresponding `div` element is removed from the DOM. 
- This results in a height increase for the map to accommodate all the undesired `div` elements.

#### Temporary Workaround:
I could fix the increased map height issue by using `overflow: hidden` or some other way. However, I just wanted to  ask if this is the intended behavior.

Is it normal for the popup component to render inside the DOM before a marker click triggers them, and should they be removed from the DOM once the popup is activated?

Thanks for your assistance!

**Comment by GrayFrost** [comment]
> Hi,
> 
> I'm encountering an issue with the `Popup` component in my Svelte app. I'm not sure if this is due to an error on my part or if it's a bug.
> 
> #### Problem:
> When the page initially loads, all components placed inside the `Popup` slots are rendered "behind" the map. When I click on a marker to open its corresponding popup, it removes the component from behind the map (and the DOM).
> 
> #### Explanation:
> Here is my map code:
> 
> ```svelte
> {#each $properties as property (property.id)}
>   <Marker
>     latlng={new LatLng(property.lat, property.lng)}
>     bind:instance={markerInstances[property.id]}
>   >
>     <Popup>
>       <PropertyCard {property} />
>     </Popup>
>   </Marker>
> {/each}
> ```
> 
> When the page loads, the HTML structure appears like this:
> 
> ```
> <div class="leaflet-container">
>   <div class="leaflet-pane leaflet-map-pane">...</div>
>   <div class="leaflet-control-container">...</div>
>   
>   <div>...</div>
> </div>
> ```
> 
> #### Observations:
> * Upon initial load, there are `div` elements corresponding to each `PropertyCard` component within the popup.
> * When I click on a marker and its popup opens, the corresponding `div` element is removed from the DOM.
> * This results in a height increase for the map to accommodate all the undesired `div` elements.
> 
> #### Temporary Workaround:
> I could fix the increased map height issue by using `overflow: hidden` or some other way. However, I just wanted to ask if this is the intended behavior.
> 
> Is it normal for the popup component to render inside the DOM before a marker click triggers them, and should they be removed from the DOM once the popup is activated?
> 
> Thanks for your assistance!

Sorry, a little busy this week, will check whether this issue is a problem this weekend. :)

**Comment by GrayFrost** [comment]
> Hi,
> 
> I'm encountering an issue with the `Popup` component in my Svelte app. I'm not sure if this is due to an error on my part or if it's a bug.
> 
> #### Problem:
> When the page initially loads, all components placed inside the `Popup` slots are rendered "behind" the map. When I click on a marker to open its corresponding popup, it removes the component from behind the map (and the DOM).
> 
> #### Explanation:
> Here is my map code:
> 
> ```svelte
> {#each $properties as property (property.id)}
>   <Marker
>     latlng={new LatLng(property.lat, property.lng)}
>     bind:instance={markerInstances[property.id]}
>   >
>     <Popup>
>       <PropertyCard {property} />
>     </Popup>
>   </Marker>
> {/each}
> ```
> 
> When the page loads, the HTML structure appears like this:
> 
> ```
> <div class="leaflet-container">
>   <div class="leaflet-pane leaflet-map-pane">...</div>
>   <div class="leaflet-control-container">...</div>
>   
>   <div>...</div>
> </div>
> ```
> 
> #### Observations:
> * Upon initial load, there are `div` elements corresponding to each `PropertyCard` component within the popup.
> * When I click on a marker and its popup opens, the corresponding `div` element is removed from the DOM.
> * This results in a height increase for the map to accommodate all the undesired `div` elements.
> 
> #### Temporary Workaround:
> I could fix the increased map height issue by using `overflow: hidden` or some other way. However, I just wanted to ask if this is the intended behavior.
> 
> Is it normal for the popup component to render inside the DOM before a marker click triggers them, and should they be removed from the DOM once the popup is activated?
> 
> Thanks for your assistance!

Hi luka, may I know which version you're using now? Maybe this feature is fixed by [#cc63719](https://github.com/GrayFrost/sveaflet/commit/cc637190fc514f4e1e59fe7b4d3231c291d554d6).
Update your Sveaflet version, and hope it helpful for you.

**Comment by lukaslovre** [comment]
Wow, I just learned that running `npm update` doesn't update your dependencies to the latest available version so i was stuck on version `0.0.4` even though i thought i was getting the latest one after running update.

Yes, it is fixed now after updating to `0.0.10`.

Sorry for the false issue and thank you for allowing me to learn something new :sweat_smile:
